pkg/controller.v1beta1/trial/trial_controller.go

```diff
@@ -244,11 +245,15 @@ func (r *ReconcileTrial) reconcileTrial(instance *trialsv1beta1.Trial) error {
 			}
 		}
 
-		// If observation is empty metrics collector doesn't finish.
-		// For early stopping metrics collector are reported logs before Trial status is changed to EarlyStopped.
+		// If observation is empty, metrics collector doesn't finish.
+		// For early stopping scenario, metrics collector will report logs before Trial status is changed to EarlyStopped.
+		// We need to requeue reconcile when the Trial is succeeded, metrics collector's type is not `Push`, and metrics are not reported.
 		if jobStatus.Condition == trialutil.JobSucceeded && instance.Status.Observation == nil {
-			logger.Info("Trial job is succeeded but metrics are not reported, reconcile requeued")
-			return errMetricsNotReported
+			if instance.Spec.MetricsCollector.Collector == nil ||
+				instance.Spec.MetricsCollector.Collector.Kind != commonapiv1beta1.PushCollector {
```

Comment:
Can you just add this condition to the L251 ? E.g.
```golang
if jobStatus.Condition == trialutil.JobSucceeded && instance.Status.Observation == nil && instance.Spec.MetricsCollector.Collector.Kind != commonapiv1beta1.PushCollector
```

Reply:
Okay for that.